{% set name = "ftfy" %}
{% set version = "5.5.0" %}
{% set sha256sum = "525ea45a871f52ddb170e66b01f35f1b3022995016c81efa305e628937b85443" %}

package:
  name: {{ name }}
  version: {{ version }}

source:
  fn: {{ name }}-{{ version }}.tar.gz
  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
  sha256: {{ sha256sum }}

build:
  number: 0
  script: python setup.py install --single-version-externally-managed --record=record.txt
  # Don't skip py37 for >=5.5.1
  skip: True  # [py2k or py>36]
  entry_points:
    - ftfy = ftfy.cli:main

requirements:
  host:
    - python
    - pytest-runner
    - setuptools

  run:
    - python
    # Required for loading package data via `pkg_resources` module.
    - setuptools
    - wcwidth

test:
  imports:
    - ftfy
    - ftfy.bad_codecs
  requires:
    - pytest
  source_files:
    - tests
  commands:
    - pytest -v tests  # [unix]
    # CLI tests fail on windows.
    - pytest -v -k "not test_cli" tests  # [win]
    - ftfy --help

about:
  home: https://github.com/LuminosoInsight/python-ftfy/
  license: MIT
  license_file: LICENSE.txt
  summary: Fixes some problems with Unicode text after the fact
  dev_url: https://github.com/LuminosoInsight/python-ftfy
  doc_url: http://ftfy.readthedocs.io/

extra:
  recipe-maintainers:
    - rmax
